packages/eslint-plugin/tests/rules/no-namespace.test.ts

```diff
@@ -49,6 +49,16 @@ declare namespace foo {
   namespace bar {
     namespace baz {}
   }
+}
+      `,
+      options: [{ allowDeclarations: true }],
+    },
+    {
+      code: `
+export declare namespace foo {
+  export namespace bar {
+    namespace baz {}
+  }
```

Comment:
I think there are few more edge cases that should be tested around declared namespaces inside a non-declared one, both exported and not:

```ts
namespace Outer {
  declare namespace Declared {
    /* export */ namespace InsideDeclared {
       // and maybe more namespaces in here
    }
  }
}
```